packages/sitecore-jss/src/layout-service.ts

```diff
@@ -126,7 +126,13 @@ export class RestLayoutService implements LayoutService {
       ? this.serviceConfig.dataFetcherResolver<LayoutServiceData>(req, res)
       : this.getDefaultFetcher<LayoutServiceData>(req, res);
 
-    return fetchRouteData(itemPath, { fetcher, ...fetchOptions });
+    return fetchRouteData(itemPath, { fetcher, ...fetchOptions }).catch((error) => {
+      if (error.response?.status === 404) {
+        return error.response.data;
```

Comment:
A comment would be helpful here to note the `response.data` shape and that it (importantly) matches the one in our GraphQL layout service.

Reply:
Added description